src/core/gltf2importer/bufferparser.cpp

```diff
@@ -104,17 +104,22 @@ bool BufferParser::parse(const QJsonArray &buffersArray, GLTF2Context *context)
     return bufferDataSize > 0;
 }
 
-QByteArray BufferParser::dataFromUri(const QString &uri, bool &success) const
+QByteArray BufferParser::dataFromUri(const QString &uri, const QDir &basePath, bool &success)
 {
-    const QString absPath = m_basePath.absoluteFilePath(uri);
-    QFile dataFile(absPath);
-    success = dataFile.open(QIODevice::ReadOnly);
-    if (success)
-        return dataFile.readAll();
-    else
-        qCWarning(kuesa) << "Failed to open" << uri;
-
-    return QByteArray();
+    if (uri.left(5).toLower() == QLatin1String("data:")) {
+        success = true;
+        return QByteArray::fromBase64(uri.toLatin1().remove(0, uri.indexOf(',') + 1));
```

Comment:
According to gltf2 spec, the URI can be a data uri following RFC 2397. 
https://tools.ietf.org/html/rfc2397
According to that, there's no need for the data uri to be base64 always. What happens if it isnt?

Reply:
I've merged the duplicate code, made more checks and warn on current limitations (ie base64 only).